nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/users/state/user-listing/user-listing.effects.ts

```diff
@@ -0,0 +1,731 @@
+
+                    if (!this.nifiCommon.isEmpty(userGroupUpdate.userGroupsAdded)) {
+                        userGroupUpdates.push(
+                            ...userGroupUpdate.userGroupsAdded
+                                .map((userGroupId: string) =>
+                                    userGroups.find((userGroup) => userGroup.id == userGroupId)
+                                )
+                                .filter((userGroup) => userGroup != null)
+                                .map((userGroup) => {
+                                    // @ts-ignore
+                                    const ug: UserGroupEntity = userGroup;
+
+                                    const users: any[] = [
+                                        ...ug.component.users.map((user) => {
+                                            return {
+                                                id: user.id
+                                            };
+                                        }),
+                                        { id: response.user.id }
+                                    ];
+
+                                    return UserListingActions.updateUserGroup({
+                                        request: {
+                                            requestId: userGroupUpdate.requestId,
+                                            revision: this.client.getRevision(userGroup),
+                                            id: ug.id,
+                                            uri: ug.uri,
+                                            userGroupPayload: {
+                                                ...ug.component,
+                                                users
+                                            }
+                                        }
+                                    });
+                                })
+                        );
+                    }
+
+                    if (!this.nifiCommon.isEmpty(userGroupUpdate.userGroupsRemoved)) {
+                        userGroupUpdates.push(
+                            ...userGroupUpdate.userGroupsRemoved
+                                .map((userGroupId: string) =>
+                                    userGroups.find((userGroup) => userGroup.id == userGroupId)
+                                )
+                                .filter((userGroup) => userGroup != null)
+                                .map((userGroup) => {
+                                    // @ts-ignore
+                                    const ug: UserGroupEntity = userGroup;
+
+                                    const users: any[] = [
+                                        ...ug.component.users
+                                            .filter((user) => user.id != response.user.id)
+                                            .map((user) => {
+                                                id: user.id;
+                                            })
```

Comment:
This doesn't construct the object correctly. It results in not being able to remove a user from a group.

```suggestion
                                            .map((user) => ({
                                                id: user.id
                                            }))
```